pymc_marketing/clv/models/beta_geo_beta_binom.py

```diff
@@ -0,0 +1,364 @@
+    """
+
+    _model_name = "BG/BB"  # Beta-Geometric Beta-Binomial Model
+    _params = ["alpha", "beta", "gamma", "delta"]
+
+    def __init__(
+        self,
+        customer_id: Union[np.ndarray, pd.Series],
+        frequency: Union[np.ndarray, pd.Series, TensorVariable],
+        recency: Union[np.ndarray, pd.Series, TensorVariable],
+        T: Union[int, np.ndarray, pd.Series, TensorVariable],
+        purchase_hyperprior: Optional[TensorVariable] = None,
+        churn_hyperprior: Optional[TensorVariable] = None,
+    ):
+        # Validate inputs before initializing model:
+
+        t_array = np.array(T)
+        if np.unique(t_array).shape[0] != 1:
+            raise ValueError(
+                "The BG/BB Model requires a static T value for all customers,"
+                " but a heterogeneous T array was provided."
+                " If using clv_summary(), try df.assign(T=df['T'].max())"
+            )
+
+        if len(np.unique(customer_id)) != len(customer_id):
+            raise ValueError(
+                "The BG/BB expects exactly one entry per customer. More than"
+                " one entry is currently provided per customer id."
+            )
+
+        super().__init__()
+
+        self._customer_id = customer_id
+        self._frequency = frequency
+        self._recency = recency
+        self._T = T
+
+        purchase_hyperprior, churn_hyperprior = self._process_priors(
+            purchase_hyperprior, churn_hyperprior
+        )
+
+        coords = {"customer_id": self._customer_id}
+        with pm.Model(coords=coords) as self.model:
+            purchase_hyper = self.model.register_rv(
+                purchase_hyperprior, name="purchase_hyperprior"
+            )
+            churn_hyper = self.model.register_rv(
+                churn_hyperprior, name="churn_hyperprior"
+            )
+
+            # Heirarchical pooling of hyperparams for beta distribution parameters.
+
+            purchase_pool = pm.Uniform(
+                "purchase_pool",
+                lower=0,
+                upper=1,
+            )
+            churn_pool = pm.Uniform(
+                "churn_pool",
+                lower=0,
+                upper=1,
+            )
+
+            alpha = pm.Deterministic("alpha", purchase_pool * purchase_hyper)
+            beta = pm.Deterministic("beta", (1.0 - purchase_pool) * purchase_hyper)
+            gamma = pm.Deterministic("gamma", churn_pool * churn_hyper)
+            delta = pm.Deterministic("delta", (1.0 - churn_pool) * churn_hyper)
+
+            # TODO: This is for plotting the population posterior, but will fail flake8
+            purchase_heterogeniety = pm.Beta(
+                "purchase_heterogeneity", alpha=alpha, beta=beta
+            )
+            # TODO: This is for plotting the population posterior, but will fail flake8
+            churn_heterogeniety = pm.Beta(
+                "churn_heterogeneity", alpha=gamma, beta=delta
+            )
```

Comment:
If these are only needed later you can create them in a dummy model and call `posterior_predictive`. This may be better because NUTS does not need to worry about these variables while sampling the ones of interest.

This is what's done for instance here: https://github.com/pymc-labs/pymc-marketing/blob/9579e1fda9077090fcaeb7d59789095915440449/pymc_marketing/clv/models/gamma_gamma.py#L51

We could have included `nu` in the model that's sampled in `fit` but it's easier for NUTS if we don't. 

When you call `posterior_predictive` with variables that are missing from the trace those will be sampled conditioned on everything else that's on the trace, leading to the same result for variables that don't contribute to the likelihood like these.

Reply:
I'll save this for a future PR involving the plotting module.